streaming/base/shuffle/py1s.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Shuffling algorithm that shuffles intra-shard in one place.
+
+This algorithm is roughly twice as fast as algorithm ``py2s``, and ever so slightly biased.
+
+Bias in this case merely refers to how we assign samples when we split shards at canonical node
+boundaries, which is non-random in this algorithm. In practice, we found this does matter to
```

Comment:
is this supposed to say "does not matter"

Reply:
https://github.com/mosaicml/streaming/pull/173